Error in HoughCircles when using UMat image input

I have been using the HoughCircles function through Python for circle detection but wanted to speed it up by making use of Transparent API. I understand that usage of UMat is necessary for that and `img` is of such type

`circles = cv2.HoughCircles(img,cv2.HOUGH_GRADIENT, 1.2, 1500)`

When I run the code, I get this error:

`hough.cpp:2254: error: (-215:Assertion failed) _circles.isMat() || _circles.isVector() in function 'HoughCircles'
`

I could find this assertion in here. 
https://github.com/opencv/opencv/blob/6ffc48769ac60d53c4bd1913eac15117c9b1c9f7/modules/imgproc/src/hough.cpp#L1660

Even with giving circles (`np.zeros((1,1,3), np.float32)`) as an input parameter, it gives the same error. From my understanding this is something internal within the function and since I have little experience with openCV codebase, I wonder if this is a purposeful feature or a bug.

--- a/modules/imgproc/src/hough.cpp
+++ b/modules/imgproc/src/hough.cpp
@@ -1728,7 +1728,6 @@ static void HoughCircles( InputArray _image, OutputArray _circles,
     }
 
     CV_Assert(!_image.empty() && _image.type() == CV_8UC1 && (_image.isMat() || _image.isUMat()));
-    CV_Assert(_circles.isMat() || _circles.isVector());
 
     if( dp <= 0 || minDist <= 0 || param1 <= 0 || param2 <= 0)
         CV_Error( Error::StsOutOfRange, "dp, min_dist, canny_threshold and acc_threshold must be all positive numbers" );

